sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -330,6 +341,10 @@ public Mono<List<Address>> getServerAddressesViaGatewayAsync(
             logger.debug("getServerAddressesViaGatewayAsync collectionRid {}, partitionKeyRangeIds {}", collectionRid,
                 JavaStreamUtils.toString(partitionKeyRangeIds, ","));
         }
+
+        // track address refresh has happened, this is only meant to be used for fault injection validation
+        request.faultInjectionRequestContext.recordAddressRefresh(forceRefresh);
```

Comment:
Can this faultInjectionRequestContext be ever null?

Reply:
it will never be null as it is always be created for each request